src/gluonts/model/tft/_estimator.py

```diff
@@ -155,161 +155,106 @@ def __init__(
-                    for name in self.static_cardinalities.keys()
-                ]
-            )
-            + [
-                AsNumpyArray(field=name, expected_ndim=1)
-                for name in chain(
-                    self.static_feature_dims.keys(),
-                    self.dynamic_cardinalities.keys(),
-                )
-            ]
-            + [
-                AsNumpyArray(field=name, expected_ndim=2)
-                for name in self.dynamic_feature_dims.keys()
-            ]
-            + [
-                AddObservedValuesIndicator(
-                    target_field=FieldName.TARGET,
-                    output_field=FieldName.OBSERVED_VALUES,
-                ),
-                AddTimeFeatures(
-                    start_field=FieldName.START,
-                    target_field=FieldName.TARGET,
-                    output_field=FieldName.FEAT_TIME,
-                    time_features=self.time_features,
-                    pred_length=self.prediction_length,
-                ),
-            ]
+        transform = AsNumpyArray(field=FieldName.TARGET, expected_ndim=1)
+
+        for name in self.static_cardinalities.keys():
+            transform += AsNumpyArray(field=name, expected_ndim=1)
+
+        for name in chain(
+            self.static_feature_dims.keys(),
+            self.dynamic_cardinalities.keys(),
+        ):
+            transform += AsNumpyArray(field=name, expected_ndim=1)
+
+        for name in self.dynamic_feature_dims.keys():
+            transform += AsNumpyArray(field=name, expected_ndim=2)
+
+        transform += AddObservedValuesIndicator(
+            target_field=FieldName.TARGET,
+            output_field=FieldName.OBSERVED_VALUES,
+        )
+
+        transform += AddTimeFeatures(
+            start_field=FieldName.START,
+            target_field=FieldName.TARGET,
+            output_field=FieldName.FEAT_TIME,
+            time_features=self.time_features,
+            pred_length=self.prediction_length,
         )
 
         if self.static_cardinalities:
-            transforms.append(
-                VstackFeatures(
-                    output_field=FieldName.FEAT_STATIC_CAT,
-                    input_fields=list(self.static_cardinalities.keys()),
-                    h_stack=True,
-                )
+            transform += HstackFeatures(
+                output_field=FieldName.FEAT_STATIC_CAT,
+                input_fields=list(self.static_cardinalities.keys()),
             )
         else:
-            transforms.extend(
-                [
-                    SetField(
-                        output_field=FieldName.FEAT_STATIC_CAT,
-                        value=[0.0],
-                    ),
-                    AsNumpyArray(
-                        field=FieldName.FEAT_STATIC_CAT, expected_ndim=1
-                    ),
-                ]
+            transform += SetArrayField(
+                output_field=FieldName.FEAT_STATIC_CAT,
+                value=[0.0],
```

Comment:
Again, would a default of `[0.0]` make sense here? I haven't seen another value yet.

Reply:
`[[0.0]]` is used at times, in general I think it's hard to have a default here because of shape